galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt

```diff
@@ -1520,85 +1508,204 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
+                      if (index == 0 && resultRow == null) {
+                        continue@loop
+                      }
+                      var i = 0
+                      while (i < lines.size) {
+                        val j = if (SKIP_FIRST_COLUMN) i + 2 else i + 1
+                        lines[i][lineCount] = queryList[j]
+                        i += 1
+                      }
+                      lineCount += 1
+                    }
+                  }
+                }
+                break
+              } catch (e: SQLException) {
+              } catch (error: java.lang.Error) {
+              } catch (rte: RuntimeException) {
+              }
+            }
+          } catch (e: Throwable) {
+            throw VRuntimeException(e)
+          }
+
+          result = if (lineCount == 0 && (newForm == null || !isNull(block!!.activeRecord))) {
+            throw VFieldException(this, MessageCode.getMessage("VIS-00001"))
+          } else {
+            val selected: Int
+            if (lineCount == 0 && newForm != null && isNull(block!!.activeRecord)) {
+              selected = newForm.add(getForm())
+            } else {
+              if (lineCount == MAX_LINE_COUNT - 1) {
+                getForm().notice(MessageCode.getMessage("VIS-00028"))
+              }
+              if (lineCount == 1 && !SHOW_SINGLE_ENTRY) {
+                selected = 0
+              } else {
+                val ld = VListDialog(list!!.columns, lines, lineCount, newForm!!)
+                selected = ld.selectFromDialog(getForm(), null, this)
+              }
+            }
+            if (selected == -1) {
+              throw VExecFailedException() // no message needed
+            } else if (selected >= lineCount) {
+              // new, retrieve it
+              var result: Any? = null
+              try {
+                while (true) {
+                  try {
+                    val auxTable = object : Table(evalListTable()) {
+                      val id = integer("ID")
+                      val column = varchar(list!!.getColumn(0)!!.column!!,
+                              list!!.getColumn(0)!!.width)
+                    }
+
+                    transaction {
+                      val queryResult = auxTable.slice(auxTable.column)
+                              .select{ auxTable.id eq selected }
+                      result = queryResult.filterIndexed { index, resultRow ->
+                        index == 1
+                      }
+                    }
+                    break
+                  } catch (e: SQLException) {
+                  } catch (error: java.lang.Error) {
+                  } catch (rte: RuntimeException) {
+                  }
+                }
+              } catch (e: Throwable) {
+                throw VRuntimeException(e)
+              }
+              result
+            } else {
+              lines[0][selected]
+            }
+          } as String?
+
```

Comment:
check kopi this is displayQueryList method so i think that you need to change colbuf to a list of column then create qrybuf and pass it to displayQueryList